crates/bevy_ecs/src/query/world_query.rs

```diff
@@ -107,13 +114,27 @@ pub unsafe trait WorldQuery {
         table: &'w Table,
     );
 
-    /// Adds any component accesses used by this [`WorldQuery`] to `access`.
+    /// Adds any component accesses to the current entity used by this [`WorldQuery`] to `access`.
     ///
     /// Used to check which queries are disjoint and can run in parallel
     // This does not have a default body of `{}` because 99% of cases need to add accesses
     // and forgetting to do so would be unsound.
     fn update_component_access(state: &Self::State, access: &mut FilteredAccess);
 
+    /// Adds any component accesses to other entities used by this [`WorldQuery`].
+    ///
+    /// This method should panic if the access would conflict with any existing access in the [`FilteredAccessSet`].
+    ///
+    /// This is used for queries to request access to entities other than the current one,
+    /// such as to read resources or to follow relations.
```

Comment:
```suggestion
    /// such as to follow relations.
```

Resources as entities isn't merged, so we shouldn't add confusing doc comments.

Reply:
> Resources as entities isn't merged, so we shouldn't add confusing doc comments.

You're right, of course, but I'm a little sad that it isn't.  The reason I started this little side quest in the first place was to fix up the safety for `AssetChanged` after resources-as-entities.